File: bindings/python/and_example.py
import numpy as np
from nnw import Model, Trainer, Activation, Optimizer, Loss, TrainerConfig

def main():
    with Model(2) as model:
        model.add_dense(1, Activation.SIGMOID)

        cfg = TrainerConfig(3000, 4, True, 0.01)
        with Trainer(model, Optimizer.ADAMW, Loss.MSE, cfg) as trainer:
            X = np.array([[0,0],[0,1],[1,0],[1,1]], dtype=np.float32)
            Y = np.array([[0],[0],[0],[1]], dtype=np.float32)
            trainer.fit(X, Y)

            preds = model.predict(X)
            print("Predictions:\n", np.round(preds, 2))

if __name__ == "__main__":
    main()

File: bindings/c#/src/Examples/AndExample.cs
using NNWrapper;

namespace Examples
{
    public static class AndExample
    {
        public static void Run()
        {
            Console.WriteLine("=== AND example ===");

            using var model = Model.Create(2);
            model.AddDense(1, Activation.SIGMOID);

            var cfg = new TrainerConfig
            {
                Epochs = 3000,
                BatchSize = 4,
                LearningRate = 0.01f
            };

            using var trainer = Trainer.Create(model, Optimizer.ADAMW, Loss.MSE, cfg);

            float[,] X =
            {
                {0, 0},
                {0, 1},
                {1, 0},
                {1, 1}
            };

            float[,] Y =
            {
                {0},
                {0},
                {0},
                {1}
            };

            trainer.Fit(X, Y);

            var preds = model.Predict(X);
            Console.WriteLine("Predictions:");
            for (int i = 0; i < preds.GetLength(0); i++)
                Console.WriteLine($"{X[i,0]} AND {X[i,1]} = {preds[i,0]:F4}");
        }
    }
}
